modules/cnf-about-topology-aware-scheduling.adoc

```diff
@@ -0,0 +1,21 @@
+// Module included in the following assemblies:
+//
+// *scalability_and_performance/cnf-topology-aware-scheduling.adoc
+
+[id="cnf-about-topology-aware-scheduling_{context}"]
+= About topology-aware scheduling
+
+NUMA architecture allows a CPU to use any available compute memory regardless of where the memory is located. This allows for increased flexibility at the expense of performance. A CPU processing a workload using memory that is outside its NUMA zone is slower that a workload processed in a single NUMA zone. Also, for I/O-constrained workloads, the network interface on a distant NUMA zone slows down how quickly information can reach the application. High-performance workloads, such as telco workloads, can’t operate to specification under these conditions. To process latency-sensitive or high-performance workloads as efficiently and quickly as possible, all the requested {product-title} compute resources (CPUs, memory, devices) should be aligned in the same NUMA zone.
+
+The default {product-title} scheduler makes scheduling decisions based on available resources of a whole compute node, not individual NUMA zones in the compute node. This can cause error conditions in the cluster when admitting a pod to a node. It can also lead to unpredictable application performance. In worst case scenarios, a complete mismatch of resource requests and kubelet policies can occur which can cause pod scheduling to fail, with the scheduler entering a failure loop. For example, runaway pod creation with `Topology Affinity Error` statuses can occur when the scheduler makes suboptimal scheduling decisions for guaranteed workloads by not knowing if the pod's requested resources are available in the compute node. Scheduling mismatch decisions can also potentially cause indefinite pod startup delays. Also, depending on the cluster state and resource allocation, poor scheduling decisions can cause extra load on the cluster because of failed startup attempts. The NUMA Resources Operator deploys a custom NUMA resources secondary scheduler to mitigate against these shortcomings.
```

Comment:
"In worst case scenarios, a complete mismatch of resource requests and kubelet policies can occur which can cause pod scheduling to fail, with the scheduler entering a failure loop. For example"
I'd just remove this part. Maybe let's rephrase like this
1. WITHOUT strict alignment request on the kubelet level, workload can be admitted with misaligned resources, causing worse/unpredictable performance
2. WITH strict alignment request on the kubelet level, but with unaware (stock, kube default) scheduler, we can have the failure loop. For example...